HARK/ConsumptionSaving/ConsIndShockModel.py

```diff
@@ -2341,6 +2346,301 @@ def calc_bounding_values(self):
         self.hNrm = hNrm
         self.MPCmin = MPCmin
         self.MPCmax = MPCmax
+        
+        
+        
+        
+    def define_distribution_grid(self, dist_mGrid=None, dist_pGrid=None, m_density = 0, num_pointsM = None,  num_pointsP = 50, max_p_fac = 7.0):
+        
+        '''
+        Defines the grid on which the distribution is defined. Stores the grid of market resources and permanent income as attributes of self.
+        Grid for normalized market resources and permanent income may be prespecified 
+        as dist_mGrid and dist_pGrid, respectively. If not then default grid is computed based off given parameters.
+        
+        Parameters
+        ----------
+        dist_mGrid : np.array
+                Prespecified grid for distribution over normalized market resources
+            
+        dist_pGrid : np.array
+                Prespecified grid for distribution over permanent income. 
+            
+        m_density: float
+                Density of normalized market resources grid. Default value is mdensity = 0.
+                Only affects grid of market resources if dist_mGrid=None.
+            
+        num_pointsM: float
+                Number of gridpoints for market resources grid.
+        
+        num_pointsP: float
+                 Number of gridpoints for permanent income. 
+                 This grid will be exponentiated by the function make_grid_exp_mult.
+                
+        max_p_fac : float
+                Factor that scales the maximum value of permanent income grid. 
+                Larger values increases the maximum value of permanent income grid.
+        
+        Returns
+        -------
+        None
+        '''  
+ 
+   
+        if not hasattr(self, "neutral_measure"):
+            self.neutral_measure = False
+            
+        if num_pointsM == None:
+            m_points = self.aXtraCount
+        else:
+            m_points = num_pointsM
+          
+        if self.cycles == 0:
```

Comment:
This could be 
`if (self.cycles == 0) or (self.cycles == 1 and self.T_cycle !=0):`
to avoid code repetition I think

Reply:
The T_cycle !=0 problem is exclusive of the cycles=0 problem so it is not unfortunately.